pkg/scheduler/framework/preemption/preemption_test.go

```diff
@@ -354,3 +366,91 @@ func TestSelectCandidate(t *testing.T) {
 		})
 	}
 }
+
+func TestFindCandidates(t *testing.T) {
+	tests := []struct {
+		name               string
+		allNodes           []*framework.NodeInfo
+		pod                *v1.Pod
+		node               *v1.Node
+		nodeStatuses       map[string]*framework.Status
+		expectedError      bool
+		expectedCandidates int
+	}{
+		{
+			name:               "No nodes available",
+			allNodes:           []*framework.NodeInfo{},
+			pod:                st.MakePod().Name("test-pod").Priority(1000).Obj(),
+			node:               st.MakeNode().Name("").Obj(),
+			nodeStatuses:       map[string]*framework.Status{},
+			expectedError:      true,
+			expectedCandidates: 0,
+		},
+		{
+			name: "Single node with no potential candidates",
+			allNodes: []*framework.NodeInfo{
+				framework.NewNodeInfo(st.MakePod().Name("victim-pod").Node("node1").Priority(midPriority).Obj()),
+			},
+			pod:  st.MakePod().Name("test-pod").Priority(1000).Obj(),
+			node: st.MakeNode().Name("node1").Obj(),
+			nodeStatuses: map[string]*framework.Status{
+				"node1": framework.NewStatus(framework.Success),
+			},
+			expectedError:      false,
+			expectedCandidates: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Mock the NodeToStatusReader
+			nodeToStatus := &FakeNodeToStatusReader{nodes: tt.nodeStatuses}
+
+			nodes := []*v1.Node{tt.node}
+
+			client := clientsetfake.NewClientset()
+			snap := internalcache.NewSnapshot([]*v1.Pod{tt.pod}, nodes)
+			_, ctx := ktesting.NewTestContext(t)
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			var objs = []runtime.Object{tt.pod}
+
+			for _, n := range nodes {
+				objs = append(objs, n)
+			}
+
```

Comment:
Can we set these operations in table-driven instead of executing them in t.Run? I think in t.Run, in order to avoid code complexity, we should focus on the code that needs to be tested, and other preparation logic should be prepared in advance or separated.

Reply:
@googs1025 ,  it follows the same style as https://github.com/liyuerich/kubernetes/blob/preemption/pkg/scheduler/framework/preemption/preemption_test.go#L314